sdk/go/pulumi-internal/netutil/netutil.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2024-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package netutil
+
+import (
+	"errors"
+	"fmt"
+	"net"
+)
+
+func isPortAvailable(port int) bool {
+	if l, err := net.Listen("tcp", fmt.Sprintf(":%d", port)); err == nil {
```

Comment:
Should we explicitly specify 127.0.0.1? See https://github.com/pulumi/pulumi/commit/25ae463915f39684419648e5ae8ee1116bde9f4e for reasoning.

Reply:
I had read this https://pkg.go.dev/net#Listen, which mentions specifying an address not being recommended.  However reading further I don't think that applies to our case at all, and especially given the reasoning in that commit we should definitely specify it.